src/transport_performance/population/rasterpop.py

```diff
@@ -261,10 +270,11 @@ def _round_population(self) -> None:
         """Round population data."""
         self._xds = np.rint(self._xds)
 
-    def _threshold_population(self, threshold: int) -> None:
+    def _threshold_population(self, threshold: Union[int, float]) -> None:
         """Threshold population data."""
         # `where()` is working differently to expectation here - it keeps
         # values above the threshold and otherwise sets them to nan.
+        _type_defence(threshold, "threshold", (int, float))
         self._xds = self._xds.where(self._xds >= threshold)
 
     def _to_geopandas(self, round: bool = False) -> None:
```

Comment:
### Suggested Change
The `._to_geopndas()` function has no type checking for `round`.
I'm aware that there is a type defence for `round` in `.get_pop()`, however if `._to_geopndas()` was to be implemented elsewhere it would lack type defences.  
Another reason to include type defences here is to be more consistent with other private functions. All other private functions in `RasterPop`, more specifically the ones used by `.get_pop()` have type defences for their parameters within the function, with `_tp_geopandas()` being the exception.

Reply:
Type checking for round does exist - it's here:

https://github.com/datasciencecampus/transport-network-performance/blob/c70dabb0a84229db423ee1c11aefe75c88eed305/src/transport_performance/population/rasterpop.py#L129-L131

It's sooner in the codebase since it's utilised before the `_to_geopandas()` call.

Happy to resolve this conversation if you are?

Reply:
After reading this again I completely agree. I''ll resolve this now 😄

Reply:
OK great! Thanks @CBROWN-ONS! All helps to save a line :) If the `_to_geopandas()` method could be useful utils function do shout though.